# ChiSquare Testing

In [1]:
import numpy as np
import scipy.stats as stats

In [2]:
# Data provided in the contingency table
observed = np.array([[50, 70], 
                     [80, 100], 
                     [60, 90], 
                     [30, 50], 
                     [20, 50]])

In [3]:
# Perform the Chi-Square test
chi2, p, dof, expected = stats.chi2_contingency(observed)

In [4]:
# Significance level
alpha = 0.05

In [5]:
critical_value = stats.chi2.ppf(1 - alpha, dof)

In [6]:
print(f"Chi-Square Statistic: {chi2}")
print(f"Degrees of Freedom: {dof}")
print(f"Critical Value: {critical_value}")
print(f"P-Value: {p}")

Chi-Square Statistic: 5.638227513227513
Degrees of Freedom: 4
Critical Value: 9.487729036781154
P-Value: 0.22784371130697179


In [8]:
if chi2 > critical_value:
    print("Reject the null hypothesis - There is an association between device type and customer satisfaction.")
else:
    print("Fail to reject the null hypothesis - No association found between device type and customer satisfaction.")

Fail to reject the null hypothesis - No association found between device type and customer satisfaction.


# Results of the Chi-Square Test
#### Chi-Square Statistic (χ²): 5.638
#### Degrees of Freedom (df): 4
#### Critical Value: 9.488
#### P-Value: 0.228
#### Expected Frequencies:
   #### Very Satisfied: 48,72
   #### Satisfied: 72,108
   #### Neutral: 60,90
   #### Unsatisfied: 32,48
   #### Very Unsatisfied: 28,42

# Conclusion
#### Chi-Square Statistic vs. Critical Value: The Chi-Square statistic (5.638) is less than the critical value (9.488).This means that we fail to reject the null hypothesis.

#### P-Value: The p-value (0.228) is greater than the significance level (0.05). This also supports the decision to fail to reject the null hypothesis.

# Final Decision
#### There is no significant association between the type of smart home device purchased (Smart Thermostat or Smart Light) and the customer's satisfaction level. The satisfaction level appears to be independent of the type of device.

# Implications
#### Mizzare Corporation might conclude that other factors, unrelated to the type of device, are influencing customer satisfaction. They may want to investigate other aspects, such as price, ease of use, or customer support, to better understand what drives customer satisfaction.